src/kex.c

```diff
@@ -98,6 +98,44 @@
             }                                                              \
     }
 
+/*
+ * Generic Diffie-Hellman computation support.
+ *
+ * DH context should be a _libssh2_bn *.
+ */
+
+#ifndef libssh2_dh_key_pair
```

Comment:
Thanks for the review.

Would'nt it be then preferable to implement both phases of Diffie-Hellman computation + constructor/destructor as crypto backend specific procedures ? This would also release the need to "publish" _libssh2_bn_mod_exp() and _libssh2_bn_rand().

If you agree, I can produce another pull request for it in a few days...